src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java/JavaLexer.g4

```diff
@@ -0,0 +1,438 @@
+LITERAL_THROWS:          'throws';
+LITERAL_TRANSIENT:       'transient';
+LITERAL_TRY:             'try';
+LITERAL_VOID:            'void';
+LITERAL_VOLATILE:        'volatile';
+LITERAL_WHILE:           'while';
+LITERAL_RECORD:          'record';
+LITERAL_YIELD:           'yield';
+LITERAL_NON_SEALED:      'non-sealed';
+LITERAL_SEALED:          'sealed';
+LITERAL_PERMITS:         'permits';
+
+// Literals
+DECIMAL_LITERAL_LONG:    ('0' | [1-9] (Digits? | '_'+ Digits)) [lL];
+DECIMAL_LITERAL:         ('0' | [1-9] (Digits? | '_'+ Digits));
+
+HEX_LITERAL_LONG:        '0' [xX] [0-9a-fA-F] ([0-9a-fA-F_]* [0-9a-fA-F])? [lL];
+HEX_LITERAL:             '0' [xX] [0-9a-fA-F] ([0-9a-fA-F_]* [0-9a-fA-F])?;
+
+OCT_LITERAL_LONG:        '0' '_'* [0-7] ([0-7_]* [0-7])? [lL];
+OCT_LITERAL:             '0' '_'* [0-7] ([0-7_]* [0-7])?;
+
+BINARY_LITERAL_LONG:     '0' [bB] [01] ([01_]* [01])? [lL];
+BINARY_LITERAL:          '0' [bB] [01] ([01_]* [01])?;
+
+
+DOUBLE_LITERAL:          (Digits '.' Digits? | '.' Digits) ExponentPart? [dD]
+             |           Digits (ExponentPart [dD] | [dD])
+             ;
+FLOAT_LITERAL:           (Digits '.' Digits? | '.' Digits) ExponentPart? [fF]?
+             |           Digits (ExponentPart [fF]? | [fF])
+             ;
+
+
+HEX_DOUBLE_LITERAL:      '0' [xX] (HexDigits '.'? | HexDigits? '.' HexDigits)
+                         [pP] [+-]? Digits [dD];
+
+HEX_FLOAT_LITERAL:       '0' [xX] (HexDigits '.'? | HexDigits? '.' HexDigits)
+                         [pP] [+-]? Digits [fFdD]?;
+
+LITERAL_TRUE:            'true';
+
+LITERAL_FALSE:           'false';
+
+CHAR_LITERAL:            '\'' (EscapeSequence | ~['\\\r\n]) '\'';
+
+STRING_LITERAL:          '"' (EscapeSequence | ~["\\\r\n])* '"';
+
+TEXT_BLOCK_LITERAL_BEGIN: '"' '"' '"' -> pushMode(TextBlock);
+
+LITERAL_NULL:            'null';
+
+// Separators
+
+LPAREN:                  '(';
+RPAREN:                  ')';
+LCURLY:                  '{';
+RCURLY:                  '}';
+LBRACK:                  '[';
+RBRACK:                  ']';
+SEMI:                    ';';
+COMMA:                   ',';
+DOT:                     '.';
+
+// Operators
+
+ASSIGN:                  '=';
+GT:                      '>';
+LT:                      '<';
+LNOT:                    '!';
+BNOT:                    '~';
+QUESTION:                '?';
+COLON:                   ':';
+EQUAL:                   '==';
+LE:                      '<=';
+GE:                      '>=';
+NOT_EQUAL:               '!=';
+LAND:                    '&&';
+LOR:                     '||';
+INC:                     '++';
+DEC:                     '--';
+PLUS:                    '+';
+MINUS:                   '-';
+STAR:                    '*';
+DIV:                     '/';
+BAND:                    '&';
+BOR:                     '|';
+BXOR:                    '^';
+MOD:                     '%';
+//SR:                    '>>'; handled in parser
+//SL:                    '<<'; handled in parser
```

Comment:
I left these here as a way to document the lack of actual SR, SL, and BSR tokens.